### DISCUSSION TITLE
How set url parameters before each test

### DISCUSSION THREAD

**Discussion by kwasilewska01** [vitest-dev/vitest#4034] [Q&A] [answered]
I migrating from jest to cutest and I have a problem with the test. I used jsdom.reconfigure({url: …}) to set some parameters from the search engine at the beginning of the test, but I don’t know how to set them in vitest? The problem is that I need to set the url, click something, and then check if the url changes, but it doesn’t work.

**Comment by simontunnat** [comment]
Did you find any solution jet?

I have the same issue and really need a fix for it.

**Comment by sheremet-va** [comment]
Did you have a separate environment? I don't think jest supports `jsdom` global variable. Currently, you can change the URL per test file (not per _test_) with this jsdoc comment at the top:

```ts
/**
 * @vitest-environment jsdom
 * @vitest-environment-options { "url": "https://example.com/" }
 */
```

**Reply by simontunnat** [reply]
That seems to work. Thank you!

But I would really like to have a way to set the URL per test. Is there anyway? :)

**Reply by sheremet-va** [reply]
There are some workarounds here: https://github.com/jestjs/jest/issues/5124

**Reply by simontunnat** [reply]
Got it to work like this:
```
    global.window = Object.create(window);

    const url = new URL("https://www.example.com");

    Object.defineProperty(window, 'location', {
        value: {
            href: url.href,
            hostname: url.hostname
        }
    });
```

That might or might not work depending on your jsdom / vitest version.

I'm currently on jsdom version 21.1.1.

**Comment by kwasilewska01** [comment, marked as answer]
But it will work when you using `<MemoryRouter />` and setting the url in the props 

But it worked when it uses code like this
```
export const renderWithRouter= (
pathToSet,
…component
) => {
const history = createMemoryHistory({ initialEntries: [pathToSet] });

render(
   <Router history={history}>
      {component}
   </Router>
)

return{
history
}
```

Now I can use it in every test 

```
const {history} = renderWithRouter(„/„, <Component />)

expect(history.location.path).toBe(„/„)
```
